### PR TITLE
[mgx-68] Feature/compound

### PR DESCRIPTION
**Pull request by vanderian**
auto compound for XYK
proxy pallet

### CODE REVIEW

File: pallets/xyk/src/lib.rs
Comment (mateuszaaa) [inline pallets/xyk/src/lib.rs:202]: i think we should keep fn docs next to the function itself. and in the overview just link to function name :thinking:

File: pallets/xyk/src/lib.rs
Comment (mateuszaaa) [inline pallets/xyk/src/lib.rs:227]: typo

File: pallets/xyk/src/lib.rs
Comment (mateuszaaa) [inline pallets/xyk/src/lib.rs:780]: consider using 'sp_runtime::Permill`

File: pallets/xyk/src/tests.rs
Comment (mateuszaaa) [inline pallets/xyk/src/tests.rs:2764]: :+1:

File: runtime/mangata-kusama/src/lib.rs
Comment (mateuszaaa) [inline runtime/mangata-kusama/src/lib.rs:1116]: when proxy call is made, whoe defines the ProxyType, is it caller?

**Review by mateuszaaa — APPROVED** [review]
lgtm, just few minor comment/questions

File: pallets/xyk/src/lib.rs
Comment (vanderian) [reply pallets/xyk/src/lib.rs:202]: might be a bit better, i was following the convention in the file, let's decide on one format and stick to it
if we decide to move the docs to funcs, lets do it in a separate PR

File: runtime/mangata-kusama/src/lib.rs
Comment (vanderian) [reply runtime/mangata-kusama/src/lib.rs:1116]: yes, you as a user/caller select the ProxyType/access level to your account

**Comment by vanderian** [comment]
blocked until #342 is merged

todo: integration test for proxies

File: pallets/xyk/src/lib.rs
Comment (goncer) [inline pallets/xyk/src/lib.rs:2600]: @vanderian  Is this expected? have you agreed this functionality with OAK and Gleb?  If I understood correctly this code, this specific line will try to mint All the Y token, but if the user has some ( for some reason ) Either this operation could fail (max_sec_to_first) , either this will mint tokens that the user would not expect. 
As a User I would expect that compound will only mint tokens from rewards, not my savings

File: runtime/mangata-kusama/src/lib.rs
Comment (vanderian) [reply runtime/mangata-kusama/src/lib.rs:1144]: @mateuszaaa see integration tests for proxy

**Review by iStrike7 — APPROVED** [review]

**Review by majusko — APPROVED** [review]

**Pull request merged**
